src/components/FilePreviewButton/index.tsx

```diff
@@ -81,7 +82,10 @@ function FilePreviewDialog(props: FilePreviewDialogProps) {
   let portalPresignedUrlRes;
   if (type == 'gds') {
     portalPresignedUrlRes = usePortalGDSPresignAPI(id, {
-      headers: { 'Content-Disposition': 'inline' },
+      headers: {
+        'Content-Disposition': 'inline',
```

Comment:
I'm thinking to explicitly define the `inline` for only certain filetype that we know we want to open `inline` and make the default of `attachment`. Say whitelisting [`pdf`, `png`, `jpg`, `html`] and the rest goes to to `attachment`. 

I'm a bit hesitant to rely on browser's behaviour to decide for preview or download. Thoughts?

I'm happy to add this if we go along this path.

Reply:
Yah, okay. Conceptually, "preview" mean that we wish to render the content inline in browser. So, the usage of `FilePreviewButton` component already bounded to this contract, implicitly. i.e. There will always be `inline` request to API.

> I'm happy to add this if we go along this path.

Understood what you mean. There might be that only limited number of file extensions are supported by this component. So, it is more like component rule.

Thinking of the other way; say, someone who use `<FilePreviewButton>` component elsewhere and, the result get ended up to attachment mode and, content gets downloaded. It is like, "_Oi, I want it to preview; pls don't download it!_" On the other hand, we can rule out upfront that this component only support sudden whitelisted file types, throw type error straight!

So, yes; go for it as component level guarding, I reckon.